src/main/java/io/vertx/ext/cluster/infinispan/impl/SubsCacheHelper.java

```diff
@@ -81,23 +87,64 @@ public SubsCacheHelper(DefaultCacheManager cacheManager, NodeSelector nodeSelect
 
   public CompletableFuture<List<RegistrationInfo>> get(String address) {
     return subsCache.get(address)
-      .thenApply(collection -> {
-        return collection.stream()
-          .map(WrappedBytes::getBytes)
-          .map(DataConverter::<RegistrationInfo>fromCachedObject)
-          .collect(toList());
+      .thenApply(remote -> {
+        List<RegistrationInfo> list;
+        int size;
+        size = remote.size();
+        Set<RegistrationInfo> local = localSubs.get(address);
+        if (local != null) {
+          synchronized (local) {
```

Comment:
Ok, that's just for multiple competing `get(address)`

Reply:
Or get + remove